### DISCUSSION TITLE
[Feature] general sorting options for views

### DISCUSSION THREAD

**Discussion by blublub** [immich-app/immich#10400] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hello

I want to suggest sorting options - I have seen requests for sorting in different ways, bot nothing "general" yet.

Sort by:
date 
size
resolution
date imported
changed/modified

all as ascending / descending

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by krzemyczek** [comment]
it would help with finding large video which take up a lot of space

**Comment by wischi-chr** [comment]
On mobile would be great too, to remove large files from the device which are already uploaded.

**Comment by bkiscool** [comment]
I would also like to see sorting by file name.

I have an album where photos are intended to be sorted by file name, but they are out of order because I can only change the display order by date created.

![image](https://github.com/user-attachments/assets/855b8e50-75b5-4519-98a0-126869e0e6a2)

**Reply by DittmannCz** [reply]
this would be great

**Comment by pippo73** [comment]
Hi, 
is there any chance this request is take in the roadmap? 
I think more filter in the search bar would be great!

**Comment by firsttobebear** [comment]
Bump, this is something I would be really interested in.

**Comment by CarlesLlobet** [comment]
+++

**Comment by bearfriend** [comment]
For searching videos, length would be helpful too. Size is not always a linear relationship due to resolution, encoding, etc.

**Comment by sadokx** [comment]
This would be amazing

**Comment by zboina** [comment]
I also kindly ask for the possibility of sorting, e.g. by date of adding a photo to immich. It would be very helpful when browsing the last added photos so that they are at the beginning of the general view (/photos)

**Comment by RockyC36** [comment]
I know that y’all already had a Discord vote on this that I missed, but being able to sort the People view would REALLY be handy

**Comment by GovSat1** [comment]
Referencing this issue: https://github.com/immich-app/immich/discussions/21092

Closed as a duplicate, but I believe Trash is a special use-case where default sort by recently deleted instead of timeline makes more sense and offers additional protection against accidental deletes.

**Comment by KatMan-59** [comment]
I would like to echo this request but in addition:

## 📌 Discussion: Sorting Filtered Results by Date/Timeline

**Current Capability:**  
Right now, Immich lets me filter assets (e.g., *“Not in any album”*), but the filtered results can’t be explicitly sorted by **date taken / timeline order**.  

**Why this matters:**  
- When organizing unfiled photos, it’s much easier to work chronologically, particularly if the photos are displayed in the same way on the main Photo screen with the timeline enabled.
- The current display order isn’t always predictable or controllable.  

**Proposal:**  
Add the ability to **sort search/filtered results** by:  
- Date Taken (asc/desc)  
- File Creation Date  

This would align filtered search with the natural “timeline” experience and make album organization much smoother.  

**Related:**  
- [Search Results organized by date (and more)](https://github.com/immich-app/immich/discussions/8377)  
- [Smart Albums / Album Membership Filtering](https://github.com/immich-app/immich/discussions/1621)

**Comment by rbhollabaugh** [comment]
I think this is related...

I have 1000's of scanned in old photos. They were scanned in with an Epson scanner. So there is no metadata. Each folder is a physical album. Each pic in each folder is named img1, img2 etc....
This is an external library. When viewing the pics they are sorted alphanumerically. Which makes no sense. They need to be sorted numerically so they show up in the order in which they were scanned.

I could not find any way to do this in immich. Did I miss something?

**Comment by kneutron** [comment]
I would really like to see Sort by Size on the Docker web interface so the largest videos and photos sort to the top, in case you want to watch something that has a long runtime or delete largest file(s) to save space

**Comment by hmronline** [comment]
Consider you have a library with pictures you were storing since first digital cameras became popular 25 years ago.
During those years pictures were moved between disks, processed with different software apps which might created additional files for thumbnails.
You might accidentally deleted some files and recurred to file recovery processes which also recovered unwanted files.
Then cellphones appeared and more pictures were extracted from different apps: not only camera, you may have pictures from social networks and messaing apps (such as WhatsApp), and therefore you know have not only pictures but all kinds of media files (wanted or not).

Now, you decide to import all those media files into immich because it seems so great to manage that mess. 
Finding duplicates works great. Finding similar pictures also works great. Searching does a great work!
But you miss some sorting features that would help you to better find those files you might want to analyze.

For example, you would like to sort by size in order to easily find thumnails, icons, and other files that were imported from external libraries, so you can review and delete them.
For that use case, you might also want to sort by resolution.

Sorting by something like the Quality Score offered by PhotoPrism would also be great.

Hope this feature is included sometime! Thanks in advance!

**Comment by jhenkel606** [comment]
Hi everyone — I wanted to highlight a real-world issue with Immich’s current sorting logic that impacts anyone importing scanned or archival photos.

**Problem**

- Immich sorts assets in the timeline (and in many album views) only by the extracted dateTaken / localDateTime. When multiple photos have the exact same timestamp, Immich does not apply any secondary sort key. As a result, the order of these photos becomes effectively random.

This happens frequently in several very common scenarios:

- Scanned photographs (FastFoto / Epson / Kodak scanners)
- Scanners embed the same EXIF date for all photos in a batch (e.g., “1986-04-01 00:00:00”).
- If you scan multiple albums from the same year/season, they all share identical timestamps.
- Restored archives where old prints or slides are assigned a date but not an exact time.
- Camera imports where the user intentionally normalizes timestamps (“set all to June 1985”).

**In all these cases, Immich displays mixed, interleaved photos in the timeline because the files have identical date values.**
This makes entire albums appear scrambled — Cruise photos mixed with Wisconsin photos, baby album mixed with school photos, etc. It also makes the timeline view unreliable for curated historical collections.

**Why filename is the logical tie-breaker**

- When timestamps are identical, filenames are the only stable, deterministic ordering, and scanners often generate perfectly sortable names, e.g.:

```
1986-spring-cruise-01.tif
1986-spring-cruise-02.tif
1986-spring-wisconsin-01.tif
1986-spring-wisconsin-02.tif
```


**Other systems (Apple Photos, Lightroom, Google Photos, DigiKam) already apply this approach:**

- Primary sort: date/time
- Secondary sort (tie-breaker): filename
- Optional tertiary: file hash or import order

This guarantees stable, predictable ordering for identical-timestamp assets.

**Proposed Solution**

Modify the server-side sort to:
`ORDER BY dateTaken ASC, originalFileName ASC`

(or DESC depending on the chosen order).

This change would:

Preserve current behavior for all normal photos

- Fix ordering for scanned/archival workflows
- Improve album/timeline consistency
- Require no changes to the data model


**Why this matters**

Immich positions itself as the ideal system for preserving personal photo history.
But many users have decades of physical photos, scans, negatives, slides, and VHS-to-JPEG frames — all of which tend to share identical timestamps.

Without a filename tie-breaker, these collections cannot be displayed in any meaningful chronological order.

**Comment by photoserver-pro** [comment]
+1 on this request.
In larger libraries, consistent and flexible sorting becomes essential, not just a “nice to have”. Being able to sort by capture date, import date, size, or modification date across views makes a big difference when navigating and maintaining growing archives.
For large setups, sorting is often used as a primary navigation and cleanup tool (e.g. finding older imports, large files, or recently changed assets). Having these options available in a generic, reusable way would significantly improve usability at scale.

**Comment by Staccado** [comment]
+1 on this request as well, sorting by size is a must have when i have duplicates of many photos with varying resolutions, I'd like to clean these up but it'll take ages manually. being able to sort all low resolution files and mass delete would be great.

**Comment by drewmerc302** [comment]
I just scanned a couple thousand photos with an Epson scanner and there's writing on the back of the photos that the scanner captures with a file_001a and file_001b naming convention where a is the front of the photo and b is the back of the photo. After scanning i had to manually correct the landscape/portrait orientation of many photos and photo backs. Doing this has ruined the sort order since the modified timestamps are now all over the place. Now none of the photo backs with description of whos in the photo is near the actual photo, it's just random backs next to random fronts with no context. 

Sorting by filename in an album would be a killer feature since my albums are a jumbled mess of unrelated backs and fronts. 

P.S. - i love immich, but this would make it even more useful.

**Comment by photoserver-pro** [comment]
@drewmerc302 Your scanned archive example is a great illustration of why filename sorting matters — and why sorting by capture date alone is not sufficient. When timestamps are unreliable or absent, filename is often the only stable ordering signal left.

This points to a broader principle: sorting is not just a convenience feature, it is a fundamental navigation primitive that should be available in every view — not only albums, but also search results, people, tags, locations, and folders. And the available sort dimensions need to reflect real-world use cases, including filename.

I've written a longer piece on how sorting, filtering, and temporal navigation need to work together as a consistent set across all views in Immich: https://github.com/immich-app/immich/discussions/27151

Would love to hear how others in this thread think about sorting beyond the album view.

**Comment by reyemb** [comment]
Would need the filename sorting as well.
